protocol/src/index.ts

```diff
@@ -651,6 +655,21 @@ export namespace Protocol {
     '|controlshtml|': readonly ['controlshtml', HTML];
     '|fieldhtml|': readonly ['fieldhtml', HTML];
     '|debug|': readonly ['debug', Message];
+    '|deinit|': readonly ['deinit'];
+    '|selectorhtml|': readonly ['selectorhtml', SelectorName, HTML];
+    '|refresh|': readonly ['refresh'];
+    '|tempnotify|': readonly [
+      'tempnotify', TempNotifyName, Message, ...[] | [Message] | [Message, string]
```

Comment:
What's going on with this type? The `...[]` is kind of confusing to me, as it the array nesting. Can this be simplified?

Reply:
It's parsed as `...([] | [Message] | [Message, string])` due to operator precedence and tuple spreading, though some style guides enforce the parentheses anyway for clarity. Could do that or manually split it into a 3-array union, your call.

Reply:
I think the parens will help at the bare minimum. 

`['tempnotify', TempNotifyName, Message] | ['tempnotify', TempNotifyName, Message, Message] | ['tempnotify', TempNotifyName, Message, Message, string]` seems the most clear (unless I'm mistaking what this type actually is?). `['tempnotify', TempNotifyName, Message, Message?, string?]` seems less precise but possibly how I might type this if I were doing it, as I'm not positive the extra precision really matters. Up to you which route you want to go, but please add parens if you want to keep the current approach :)

Reply:
The `?` suffix allows for `undefined` and/or skipping the tuple element, so that might not be ideal. Probably best to expand it into the type union then.

Will do this for `|expire|` as well.

Reply:
Yeah, it's less accurate, but we're also trying to put super rigorous types on top of something which is basically just raw strings on the PS side, so I'm usually not to bothered about dropping an `?` for convenience because PS is more likely to break things in the protocol than the `undefined` is going to cause problems for people :). 

Technically its also the case that for like `['expire']` you can still index at 1 and get `undefined`, so really the main difference with `?` is that you're basically implying the array should be length 2 with the second element occasionally `undefined`, though if I really wanted to be explicit about that I'd probably go `Message | undefined` (its also the case that nothing in the protocol can be `undefined` because we can't translate `undefined` over the wire).

**tl;dr** pedantically yes, the `?` implies some things different than the union, but in practice I'm not too hot and bothered by it in the context of the PS protocol :)